src/server/generic_family.cc

```diff
@@ -207,14 +202,11 @@ class RestoreArgs {
 
 [[nodiscard]] bool RestoreArgs::UpdateExpiration(int64_t now_msec) {
   if (HasExpiration()) {
-    auto new_ttl = CalculateExpirationTime(!abs_time_, abs_time_, expiration_, now_msec);
-    if (new_ttl > kMaxExpireDeadlineSec * 1000) {
+    int64_t ttl = abs_time_ ? expiration_ - now_msec : expiration_;
```

Comment:
At some point we should really consider `chrono` for anything related to time. `expiration` could just be `std::chrono::milliseconds`. That's a personal preference though, I am not advocating this heavily. 

Moreover, consider making `ttl` a `const`. I am not advocating this heavily but my inner ocd for const everything is hard to ignore and not write a comment :eyes:

Reply:
Yes, same as with using mixed int/uint. Its a little mess but it's everywhere

Reply:
I don't see any reason making local variables in a 5 line function ever const, its visual clutter. What does it guard against? Accidentally assigning it and not the member variable?